src/registrar/tests/test_views.py

```diff
@@ -2229,6 +2229,37 @@ def test_application_withdraw(self):
         home_page = self.app.get("/")
         self.assertContains(home_page, "Withdrawn")
 
+    def test_application_withdraw_no_permissions(self):
+        """Can't withdraw applications as a restricted user."""
+        self.user.status = User.RESTRICTED
+        self.user.save()
+        application = completed_application(status=DomainApplication.ApplicationStatus.SUBMITTED, user=self.user)
+        application.save()
+
+        home_page = self.app.get("/")
+        self.assertContains(home_page, "city.gov")
+        # click the "Manage" link
+        detail_page = home_page.click("Manage", index=0)
+        self.assertContains(detail_page, "city.gov")
+        self.assertContains(detail_page, "city1.gov")
+        self.assertContains(detail_page, "Chief Tester")
+        self.assertContains(detail_page, "testy@town.com")
+        self.assertContains(detail_page, "Admin Tester")
+        self.assertContains(detail_page, "Status:")
+        # Restricted user trying to withdraw results in 403 error
+        with less_console_noise():
+            for url_name in [
+                "application-withdraw-confirmation",
+                "application-withdrawn",
+            ]:
+                with self.subTest(url_name=url_name):
+                    page = self.client.get(reverse(url_name, kwargs={"pk": application.pk}))
+                    self.assertEqual(page.status_code, 403)
+
+        # reset user status from restricted to unrestricted
+        self.user.status = None
+        self.user.save()
```

Comment:
Thank you for pointing that out! I added it as a precaution but didn't realized that our tearDown already deletes whatever user exists and setUp creates a new user who has default all permissions. Just verified that it works without this reset (and it isn't needed since the user for this test is deleted after this test finishes running anyway) and removed these lines - thank you for catching that!